gap/congruences/cong.gi

```diff
@@ -390,17 +424,69 @@ function(class, elm)
   return EquivalenceClassOfElementNC(cong, Representative(class) * elm);
 end);
 
+SEMIGROUPS.GenericCongLookup := function(cong)
+  local S, lookup, class, nr, elm;
+  S := Range(cong);
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: SEMIGROUPS.GenericCongLookup: usage,\n",
+                  "<cong> must be over a finite semigroup,");
+  fi;
+  lookup := [1 .. Size(S)];
+  for class in NonTrivialEquivalenceClasses(cong) do
+    nr := PositionCanonical(S, Representative(class));
```

Comment:
And it would be better to do `TryNextMethod();` instead of returning an error from a cryptic hidden function.